boot/boot.go

```diff
@@ -141,15 +141,35 @@ func Kernel(s snap.PlaceInfo, t snap.Type, dev snap.Device) BootKernel {
 	return trivial{}
 }
 
-func applicable(s snap.PlaceInfo, t snap.Type, dev snap.Device) bool {
-	if dev.Classic() {
+// SnapTypeAffectsBootForDev finds out if a snap type participates in the
+// boot for a given device
+func SnapTypeAffectsBootForDev(t snap.Type, dev snap.Device) bool {
```

Comment:
ForDev is strange, we have other things taking Device here. SnapTypeParticipatesInBoot seems a name more consistent with other things here.

Reply:
Changed